components/cards/ClaimAction.tsx

```diff
@@ -13,78 +21,172 @@ export type ClaimActionProps = SidebarProps & {
 
   //   updateActionData({ transactionSubtotal: _transactionSubtotal });
   // }, [displayNetworkFeePayment]);
 
   async function _claim() {
-    // FIXME: Update for streaming payments
-    // updateActionData({ isActing: true });
-    // let baseCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase1Coord.x,
-    //   claimBase1Coord.y
-    // );
-    // let destCoord = GeoWebCoordinate.make_gw_coord(
-    //   claimBase2Coord.x,
-    //   claimBase2Coord.y
-    // );
-    // let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
-    //   (v) => {
-    //     return BigNumber.from(v.toString(10));
-    //   }
-    // );
-    // if (path.length == 0) {
-    //   path = [BigNumber.from(0)];
-    // }
-    // const newValue = calculateWeiSubtotalField(displayNewForSalePrice);
-    // const newContributionRate = fromValueToRate(
-    //   newValue,
-    //   perSecondFeeNumerator,
-    //   perSecondFeeDenominator
-    // );
-    // const resp = await claimerContract.claim(
-    //   account,
-    //   baseCoord.toString(10),
-    //   path,
-    //   newContributionRate,
-    //   {
-    //     from: account,
-    //     value: calculateWeiSubtotalField(displayNetworkFeePayment),
-    //   }
-    // );
-    // const receipt = await resp.wait();
-    // const filter = claimerContract.filters.ParcelClaimed(null, null);
-    // const res = await claimerContract.queryFilter(
-    //   filter,
-    //   receipt.blockNumber,
-    //   receipt.blockNumber
-    // );
-    // let licenseId = res[0].args[0];
-    // return licenseId;
-    return "";
+    const baseCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase1Coord.x,
+      claimBase1Coord.y
+    );
+    const destCoord = GeoWebCoordinate.make_gw_coord(
+      claimBase2Coord.x,
+      claimBase2Coord.y
+    );
+    let path = GeoWebCoordinate.make_rect_path(baseCoord, destCoord).map(
+      // FIXME: it should be 'BN' type, waiting for js-geo-web-coordinate upgrade.
+      (v: any) => {
+        return BigNumber.from(v.toString(10));
+      }
+    );
+    if (path.length == 0) {
+      path = [BigNumber.from(0)];
+    }
+
+    if (!displayNewForSalePrice) {
+      throw new Error(
+        `displayNewForSalePrice is invalid: ${displayNewForSalePrice}`
+      );
+    }
+
+    console.log(actionData);
+
+    let claimData;
+
+    let actionDataToPassInUserData;
+
+    let userData;
+
+    let txn;
+
+    const filter = claimerContract.filters.ParcelClaimed(null, null);
+    const parcels = await claimerContract.queryFilter(filter);
+    const existingParcels = parcels.length > 0;
+    console.log("existing parcels: ", parcels);
+
+    const sf = await sfInstance(NETWORK_ID, provider);
+
+    const ethx = await sf.loadSuperToken(paymentTokenAddress);
+
+    const approveOperation = await ethx.approve({
+      receiver: auctionSuperApp.address,
+      amount: ethers.utils.parseEther(displayNewForSalePrice).toString(),
+    });
+
+    const newFlowRate = ethers.utils
```

Comment:
The flow rate should be equal to `networkFeeRatePerSecond` in `ActionForm` [here](https://github.com/Geo-Web-Project/cadastre/pull/126/files#diff-958d7daaf87d1b26c5e80206f1032e04b0cb6129727438f9e95eb1c885d63914R110)